test/suite/test_import01.py

```diff
@@ -74,16 +74,18 @@ def check(self, uri, keys, values):
             else:
                 self.check_record(uri, keys[i], values[i])
 
-    # We know the ID can be different between configs, so just remove it from comparison.
-    # Everything else should be the same.
+    # The ID and checkpoint information can be different between configs, remove it. Everything else
+    # should be the same.
     def config_compare(self, aconf, bconf):
         a = re.sub('id=\d+,?', '', aconf)
-        a = (re.sub('\w+=\(.*?\)+,?', '', a).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', a))
+        a = re.sub('checkpoint=\(.*?\)+,?', '', a)
         b = re.sub('id=\d+,?', '', bconf)
-        b = (re.sub('\w+=\(.*?\)+,?', '', b).strip(',').split(',') +
```

Comment:
The previous code was stripping all the subconfigs (ignoring my other comment on greedy regexes), sorting the simple configs, and then appending all the subconfigs at the end before sorting. 
With the new code that doesn't do this we'll end up splitting subconfigs like `version=(major=2,minor=1),` into two items `version=(major` and `minor=1)` for comparison. It's still correct, but seems messy.

Reply:
I agree, I reverted the regex.